src/cascade/core/db.py

```diff
@@ -0,0 +1,55 @@
+from contextlib import contextmanager
+
+try:
+    from db_tools import ezfuncs
+except ImportError:
+
+    class DummyEZFuncs:
+        def __getattr__(self, name):
+            raise ImportError(f"Required package db_tools not found")
+
+    ezfuncs = DummyEZFuncs()
+
+
+@contextmanager
+def cursor(execution_context=None, database=None):
+    """A context manager which exposes a database cursor connected to the database specified by
+    the execution_context. The cursor will be closed when the manager exits and if it exits without
```

Comment:
The first sentence is making me wonder if database is needed, how it's used.

Reply:
Yeah, I agree.